server/channels/app/import_functions_test.go

```diff
@@ -748,809 +748,917 @@ func TestImportImportUser(t *testing.T) {
+	t.Run("import a valid user in dry-run", func(t *testing.T) {
+		data := imports.UserImportData{
+			Username: model.NewPointer(model.NewUsername()),
+			Email:    model.NewPointer(model.NewId() + "@example.com"),
+		}
+		appErr := th.App.importUser(th.Context, &data, true)
+		require.Nil(t, appErr, "Should have succeeded to import valid user.")
 
-	// Check that no more users are in the DB.
-	userCount3, err := th.App.Srv().Store().User().Count(model.UserCountOptions{
-		IncludeDeleted:     true,
-		IncludeBotAccounts: false,
+		// Check that no more users are in the DB.
+		userCountCurrent, err := th.App.Srv().Store().User().Count(model.UserCountOptions{
+			IncludeDeleted:     true,
+			IncludeBotAccounts: false,
+		})
+		require.NoError(t, err, "Failed to get user count.")
+		assert.Equal(t, userCount, userCountCurrent, "Unexpected number of users")
 	})
-	require.NoError(t, err, "Failed to get user count.")
-	assert.Equal(t, userCount, userCount3, "Unexpected number of users")
 
-	// Do an invalid user in apply mode.
-	data = imports.UserImportData{
-		Username: model.NewPointer(model.NewUsername()),
-	}
-	err = th.App.importUser(th.Context, &data, false)
-	require.Error(t, err, "Should have failed to import invalid user.")
+	t.Run("import an invalid user in apply mode", func(t *testing.T) {
+		data := imports.UserImportData{
+			Username: model.NewPointer(model.NewUsername()),
+		}
+		appErr := th.App.importUser(th.Context, &data, false)
+		require.NotNil(t, appErr, "Should have failed to import invalid user.")
 
-	// Check that no more users are in the DB.
-	userCount4, err := th.App.Srv().Store().User().Count(model.UserCountOptions{
-		IncludeDeleted:     true,
-		IncludeBotAccounts: false,
+		// Check that no more users are in the DB.
+		userCountCurrent, err := th.App.Srv().Store().User().Count(model.UserCountOptions{
+			IncludeDeleted:     true,
+			IncludeBotAccounts: false,
+		})
+		require.NoError(t, err, "Failed to get user count.")
+		assert.Equal(t, userCount, userCountCurrent, "Unexpected number of users")
 	})
-	require.NoError(t, err, "Failed to get user count.")
-	assert.Equal(t, userCount, userCount4, "Unexpected number of users")
 
-	// Do a valid user in apply mode.
-	username := model.NewUsername()
-	testsDir, _ := fileutils.FindDir("tests")
-	data = imports.UserImportData{
-		Avatar: imports.Avatar{
-			ProfileImage: model.NewPointer(filepath.Join(testsDir, "test.png")),
-		},
-		Username:  &username,
-		Email:     model.NewPointer(model.NewId() + "@example.com"),
-		Nickname:  model.NewPointer(model.NewId()),
-		FirstName: model.NewPointer(model.NewId()),
-		LastName:  model.NewPointer(model.NewId()),
-		Position:  model.NewPointer(model.NewId()),
-	}
-	appErr = th.App.importUser(th.Context, &data, false)
-	require.Nil(t, appErr, "Should have succeeded to import valid user.")
+	t.Run("import a valid user in apply mode", func(t *testing.T) {
+		username := "A" + model.NewUsername()[1:]
```

Comment:
I just replace the first letter to be an uppercase letter and not change the length of it (in case if the limits change at some point, this ensures that we still follow the `model.NewUsername()` length here.

Reply:
Right, but my question is more: why?